pkg/resolver/client/ens/ens_test.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package ens_test
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethersphere/bee/pkg/resolver/client/ens"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+func TestNewClient(t *testing.T) {
+	cl := ens.NewClient()
+	if cl.Endpoint != "" {
+		t.Errorf("expected no endpoint set")
+	}
+}
+
+func TestConnect(t *testing.T) {
+	ep := "test"
+
+	t.Run("no dial func error", func(t *testing.T) {
+		c := ens.NewClient(
+			ens.WithDialFunc(nil),
+		)
+		err := c.Connect(ep)
+		defer c.Close()
+		if !errors.Is(err, ens.ErrNotImplemented) {
```

Comment:
i don't understand the semantics of `not implemented` as an error for no dial function.... this is confusing